mantidproject/mantid
# Fix shrinking instrument view image on high-pixel density displays

**Description of work.**

Fixes issues with scaling of the instrument view images depending on the pixel density of the monitor display.

**To test:**

This needs to be tested on a mac with retina screen and an additional lower-resolution screen connected.
Use the following script to determine this information: 

```python
from qtpy.QtWidgets import QApplication
for screen in QApplication.screens():
    print(screen.name())
    print(screen.devicePixelRatio())
```

It should print 2.0 for a retina display and 1.0 for a standard external screen.
If all values are the same then the test of moving the window between screens will not be valid.

All tests will need a workspace with an instrument. Pick any file from the standard test data.

**1. Open on Retina Screen**

* Move workbench to the retina screen
* Click show instrument and check the instrument image is full size in the viewport
* Change to one of the unwrapped views, i.e `CylinderY` and the image should still be fullsize in the viewport

**2. Open on the Standard Def Screen**

* Close any instrument view windows
* Move workbench to the standard definition screen
* Click show instrument and check the instrument image is full size in the viewport
* Change to one of the unwrapped views, i.e `CylinderY` and the image should still be fullsize in the viewport

**3. Move Window Between Screens**

* Close any instrument view windows
* Click show instrument and check the instrument image is full size in the viewport
* Drag the window to the retina screen and check that one the drag is complete the image is still full size in the viewport
* Drag the window back and check again.
* Do the above for the unwrapped view too.

Fixes #34081

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
